uniffi/src/bindings/kotlin/templates/ObjectTemplate.kt

```diff
@@ -1,25 +1,34 @@
 
-class {{ obj.name()|class_name_kt }}(handle: Long) {
-    private var handle: AtomicLong = AtomicLong(handle)
-    {%- for cons in obj.constructors() %}
+class {{ obj.name()|class_name_kt }}(
+    internal val handle: Long
+) : FFIObject() {
+
+    {% for cons in obj.constructors() %}
     constructor({% call kt::arg_list_decl(cons.arguments()) -%}) :
         this({% call kt::to_rs_call(cons) %})
     {%- endfor %}
 
-    // XXX TODO: destructors or equivalent.
+    override fun destroy() {
+        super.destroy()
+        {% call kt::to_rs_call_with_prefix("this.handle", obj.reserved_object_free()) %}
```

Comment:
Are we are deliberately calling `super.destroy()` first so that we "poison" the Kotlin side of the object, as a guard against potential concurrency issues. I expect that the underlying handlemap will prevent concurrency issues in practice, but still, if the ordering here is important it may be worth a comment to that effect.